### PR TITLE
POM updates to fix issues with netty, log4j and solr-core dependencies

### LINKED ISSUE #310: Remove redundant dependency on org.apache.solr:solr-core
When you want to use both solr and embeded lucene index backend (e.g. one for production, other for running tests), you need to specify two components on your maven/gradle dependencies list:

```
dependencies {
    compile 'org.janusgraph:janusgraph-solr:0.1.0'    // adds redundant transitive depency to lucene 5.x
    compile 'org.janusgraph:janusgraph-lucene:0.1.0'  // uses lucene 4.x
}
```

But then the embeded lucene stops working (you get AbstractMethod errors) due to version discrepancies between lucene 4.x and 5.x.

Inspecting the dependency tree shows the problem:

**+--- org.janusgraph:janusgraph-solr:0.1.0**
|    +--- org.janusgraph:janusgraph-core:0.1.0 (*)
|    +--- org.apache.solr:solr-solrj:5.2.1
|    |    +--- commons-io:commons-io:2.4
|    |    +--- org.apache.httpcomponents:httpclient:4.4.1 (*)
|    |    +--- org.apache.httpcomponents:httpcore:4.4.1
|    |    +--- org.apache.httpcomponents:httpmime:4.4.1
|    |    +--- org.apache.zookeeper:zookeeper:3.4.6
|    |    +--- org.codehaus.woodstox:stax2-api:3.1.4
|    |    +--- org.codehaus.woodstox:woodstox-core-asl:4.4.1
|    |    +--- org.noggit:noggit:0.6
|    |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.25
**|    +--- org.apache.solr:solr-core:5.2.1
|    |    +--- org.apache.lucene:lucene-analyzers-common:5.2.1
|    |    |    \--- org.apache.lucene:lucene-core:5.2.1**
|    |    +--- org.apache.lucene:lucene-analyzers-kuromoji:5.2.1
|    |    +--- org.apache.lucene:lucene-analyzers-phonetic:5.2.1
|    |    +--- org.apache.lucene:lucene-backward-codecs:5.2.1
|    |    +--- org.apache.lucene:lucene-codecs:5.2.1
|    |    +--- org.apache.lucene:lucene-core:5.2.1
|    |    +--- org.apache.lucene:lucene-expressions:5.2.1
|    |    +--- org.apache.lucene:lucene-grouping:5.2.1
|    |    +--- org.apache.lucene:lucene-highlighter:5.2.1
|    |    +--- org.apache.lucene:lucene-join:5.2.1
|    |    +--- org.apache.lucene:lucene-memory:5.2.1
|    |    +--- org.apache.lucene:lucene-misc:5.2.1
|    |    +--- org.apache.lucene:lucene-queries:5.2.1
|    |    +--- org.apache.lucene:lucene-queryparser:5.2.1
|    |    |    +--- org.apache.lucene:lucene-core:5.2.1
|    |    |    +--- org.apache.lucene:lucene-queries:5.2.1
|    |    |    \--- org.apache.lucene:lucene-sandbox:5.2.1
|    |    +--- org.apache.lucene:lucene-spatial:5.2.1
|    |    |    +--- org.apache.lucene:lucene-core:5.2.1
|    |    |    +--- org.apache.lucene:lucene-misc:5.2.1
|    |    |    +--- org.apache.lucene:lucene-queries:5.2.1
|    |    |    \--- com.spatial4j:spatial4j:0.4.1
|    |    +--- org.apache.lucene:lucene-suggest:5.2.1
|    |    +--- org.apache.solr:solr-solrj:5.2.1 (*)
|    |    +--- com.carrotsearch:hppc:0.5.2 -> 0.7.1
|    |    +--- com.google.guava:guava:14.0.1 -> 18.0
|    |    +--- com.google.protobuf:protobuf-java:2.5.0
|    |    +--- com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru:1.2 -> 1.3
|    |    +--- com.tdunning:t-digest:3.1
|    |    +--- commons-cli:commons-cli:1.2
|    |    +--- commons-codec:commons-codec:1.10
|    |    +--- commons-collections:commons-collections:3.2.1 -> 3.2.2
|    |    +--- commons-configuration:commons-configuration:1.6 -> 1.10 (*)
|    |    +--- commons-fileupload:commons-fileupload:1.2.1
|    |    +--- commons-io:commons-io:2.4
|    |    +--- commons-lang:commons-lang:2.6
|    |    +--- dom4j:dom4j:1.6.1 (*)
|    |    +--- it.unimi.dsi:fastutil:6.5.11
|    |    +--- javax.servlet:javax.servlet-api:3.1.0
|    |    +--- joda-time:joda-time:2.2
|    |    +--- log4j:log4j:1.2.17
|    |    +--- net.agkn:hll:1.6.0
|    |    +--- org.antlr:antlr-runtime:3.5 -> 3.5.2
|    |    +--- org.apache.hadoop:hadoop-annotations:2.6.0
|    |    +--- org.apache.hadoop:hadoop-auth:2.6.0
|    |    +--- org.apache.hadoop:hadoop-common:2.6.0
|    |    +--- org.apache.hadoop:hadoop-hdfs:2.6.0
|    |    +--- org.apache.httpcomponents:httpclient:4.4.1 (*)
|    |    +--- org.apache.httpcomponents:httpcore:4.4.1
|    |    +--- org.apache.httpcomponents:httpmime:4.4.1
|    |    +--- org.codehaus.woodstox:stax2-api:3.1.4
|    |    +--- org.codehaus.woodstox:woodstox-core-asl:4.4.1
|    |    +--- org.eclipse.jetty:jetty-continuation:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-deploy:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-http:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-io:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-jmx:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-rewrite:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-security:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-server:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-servlet:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-servlets:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-util:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-webapp:9.2.10.v20150310
|    |    +--- org.eclipse.jetty:jetty-xml:9.2.10.v20150310
|    |    +--- org.htrace:htrace-core:3.0.4
|    |    +--- org.noggit:noggit:0.6
|    |    +--- org.ow2.asm:asm:4.1
|    |    +--- org.ow2.asm:asm-commons:4.1
|    |    +--- org.restlet.jee:org.restlet:2.3.0
|    |    +--- org.restlet.jee:org.restlet.ext.servlet:2.3.0
|    |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.25
|    +--- javax.servlet:servlet-api:2.5
|    +--- com.spatial4j:spatial4j:0.4.1
|    \--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
**+--- org.janusgraph:janusgraph-lucene:0.1.0
|    +--- org.janusgraph:janusgraph-core:0.1.0 (*)
|    +--- org.apache.lucene:lucene-core:4.10.4 -> 5.2.1**
|    +--- org.apache.lucene:lucene-analyzers-common:4.10.4 -> 5.2.1 (*)
|    +--- org.apache.lucene:lucene-queryparser:4.10.4 -> 5.2.1 (*)
|    +--- org.apache.lucene:lucene-spatial:4.10.4 -> 5.2.1 (*)
|    \--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25

The  org.apache.lucene:lucene-analyzers-common:4.10.4 needed by embeded lucene gets promoted to version 5.2.1 by automatic version conflict resolution.

To workaround the issue you must explicitly exlcude solr-core from your deps:

```
    compile ("org.janusgraph:janusgraph-solr:0.10") {
        exclude group: 'org.apache.solr', module: 'solr-core'
    }
    compile "org.janusgraph:janusgraph-lucene:0.10"
```

which gives the correct (much shorter) deps tree:

**--- org.janusgraph:janusgraph-solr:0.1.0
|    +--- org.janusgraph:janusgraph-core:0.1.0 (*)
|    +--- org.apache.solr:solr-solrj:5.2.1**
|    |    +--- commons-io:commons-io:2.4
|    |    +--- org.apache.httpcomponents:httpclient:4.4.1 (*)
|    |    +--- org.apache.httpcomponents:httpcore:4.4.1
|    |    +--- org.apache.httpcomponents:httpmime:4.4.1
|    |    +--- org.apache.zookeeper:zookeeper:3.4.6
|    |    +--- org.codehaus.woodstox:stax2-api:3.1.4
|    |    +--- org.codehaus.woodstox:woodstox-core-asl:4.4.1
|    |    +--- org.noggit:noggit:0.6
|    |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.25
|    +--- javax.servlet:servlet-api:2.5
|    +--- com.spatial4j:spatial4j:0.4.1
|    \--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
**+--- org.janusgraph:janusgraph-lucene:0.1.0**
|    +--- org.janusgraph:janusgraph-core:0.1.0 (*)
**|    +--- org.apache.lucene:lucene-core:4.10.4**
|    +--- org.apache.lucene:lucene-analyzers-common:4.10.4
|    |    \--- org.apache.lucene:lucene-core:4.10.4
|    +--- org.apache.lucene:lucene-queryparser:4.10.4
|    |    +--- org.apache.lucene:lucene-core:4.10.4
|    |    +--- org.apache.lucene:lucene-queries:4.10.4
|    |    \--- org.apache.lucene:lucene-sandbox:4.10.4
|    +--- org.apache.lucene:lucene-spatial:4.10.4
|    |    +--- org.apache.lucene:lucene-core:4.10.4
|    |    +--- org.apache.lucene:lucene-queries:4.10.4
|    |    \--- com.spatial4j:spatial4j:0.4.1
|    \--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25

The solr-core dep is redundant, since solr-client is enough to communicate with the server and we do not need to pollute the deps with actual server implementation jars. 

Please fix this in your maven build config.

### LINKED ISSUE #468: CQL Backend Fails to Init
When I was trying to get a scala application working with only the following jars:
```
"org.janusgraph" % "janusgraph-core" % "0.2.0-SNAPSHOT",
"org.janusgraph" % "janusgraph-cql" % "0.2.0-SNAPSHOT",
"org.janusgraph" % "janusgraph-es" % "0.2.0-SNAPSHOT",
```
I got the following error:
```
[info]   java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.cql.CQLStoreManager
[info]   at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:69)
[info]   at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:477)
[info]   at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:409)
[info]   at org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.<init>(GraphDatabaseConfiguration.java:1353)
[info]   at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:107)
[info]   at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:75)
[info]   at SimpleSpec.$anonfun$new$1(SimpleSpec.scala:13)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   ...
[info]   Cause: java.lang.reflect.InvocationTargetException:
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[info]   at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:58)
[info]   at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:477)
[info]   at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:409)
[info]   at org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.<init>(GraphDatabaseConfiguration.java:1353)
[info]   at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:107)
[info]   at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:75)
[info]   ...
[info]   Cause: java.lang.NoClassDefFoundError: io/netty/util/Timer
[info]   at com.datastax.driver.core.Configuration$Builder.build(Configuration.java:317)
[info]   at com.datastax.driver.core.Cluster$Builder.getConfiguration(Cluster.java:1282)
[info]   at com.datastax.driver.core.Cluster.<init>(Cluster.java:113)
[info]   at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:178)
[info]   at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:1299)
[info]   at org.janusgraph.diskstorage.cql.CQLStoreManager.initializeCluster(CQLStoreManager.java:248)
[info]   at org.janusgraph.diskstorage.cql.CQLStoreManager.<init>(CQLStoreManager.java:145)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[info]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[info]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[info]   ...
[info]   Cause: java.lang.ClassNotFoundException: io.netty.util.Timer
[info]   at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
[info]   at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[info]   at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[info]   at com.datastax.driver.core.Configuration$Builder.build(Configuration.java:317)
[info]   at com.datastax.driver.core.Cluster$Builder.getConfiguration(Cluster.java:1282)
[info]   at com.datastax.driver.core.Cluster.<init>(Cluster.java:113)
[info]   at com.datastax.driver.core.Cluster.buildFrom(Cluster.java:178)
[info]   at com.datastax.driver.core.Cluster$Builder.build(Cluster.java:1299)
[info]   at org.janusgraph.diskstorage.cql.CQLStoreManager.initializeCluster(CQLStoreManager.java:248)
[info]   at org.janusgraph.diskstorage.cql.CQLStoreManager.<init>(CQLStoreManager.java:145)
[info]   ...
```

Once I added the `"io.netty" % "netty-all" % "4.1.14.Final"` dependency the code ran fine. Not sure if this is a bug since I'm pulling the jars in piecemeal but saw that the cassandra-driver is excluding netty and probably the cause of my problem:
```xml
<dependency>
  <groupId>com.datastax.cassandra</groupId>
  <artifactId>cassandra-driver-core</artifactId>
  <version>${cassandra-driver.version}</version>
  <exclusions>
    <!-- conflicts with netty-all from org.apache.hbase:hbase-client -->
    <exclusion>
      <groupId>io.netty</groupId>
      <artifactId>netty-handler</artifactId>
    </exclusion>
    </exclusions>
</dependency>
```

If someone points me in the right direction on how to fix this I'd gladly open a PR but my maven foo is weak.

### LINKED ISSUE #517: SLF4j binding issue resulted in no internal logging
Problem:
I am using JanusGraph 0.1.1. And using logback:

       <logger name="org.janusgraph" level="DEBUG" additivity="false">

<appender-ref ref="CONSOLE" />

</logger>

<logger name="org.apache" level="DEBUG" additivity="false">

<appender-ref ref="CONSOLE" />

</logger>

        <root level="debug">

<appender-ref ref="CONSOLE" />

</root>

This result in following output only:
```
15:28:23,108  INFO ReflectiveConfigOptionLoader:173 - Loaded and initialized config classes: 9 OK out of 12 attempts in PT0.018S
15:28:23,171  INFO Reflections:224 - Reflections took 38 ms to scan 1 urls, producing 12 keys and 25 values 
15:28:23,244  INFO CassandraThriftStoreManager:628 - Closed Thrift connection pooler.
15:28:23,251  INFO GraphDatabaseConfiguration:1563 - Generated unique-instance-id=0a96cb391964-MacBook-Pro-3-local1
15:28:23,263  INFO Backend:463 - Configuring index [search]
15:28:23,355  INFO plugins:151 - [Solarr] loaded [], sites []
15:28:23,875  INFO ElasticSearchIndex:368 - Configured remote host: 127.0.0.1 : 9300
15:28:24,043  INFO Backend:178 - Initiated backend operations thread pool of size 8
15:28:24,165  INFO KCVSLog:744 - Loaded unidentified ReadMarker start time 2017-09-13T09:58:24.152Z into org.janusgraph.diskstorage.log.kcvs.KCVSLog$MessagePuller@7fcff1b9
15:28:25,409  INFO CassandraThriftStoreManager:628 - Closed Thrift connection pooler.
```

Solution:
As per SLF4j:
Embedded components such as libraries or frameworks should not declare a dependency on any SLF4J binding but only depend on slf4j-api. When a library declares a compile-time dependency on a SLF4J binding, it imposes that binding on the end-user, thus negating SLF4J's purpose. When you come across an embedded component declaring a compile-time dependency on any SLF4J binding, please take the time to contact the authors of said component/library and kindly ask them to mend their ways.

Request to exclude following dependency 

```
<groupId>org.slf4j</groupId>
<artifactId>slf4j-log4j12</artifactId>
```

from JanusGraph.

```
<groupId>org.janusgraph</groupId>
<artifactId>janusgraph-es</artifactId>
<artifactId>janusgraph-cassandra</artifactId>
<artifactId>gremlin-driver</artifactId>
```

Logging will start working.

### PR DESCRIPTION
**Pull request by sjudeng**
Fixes #517 
Fixes #468 
Fixes #310

### CODE REVIEW

File: janusgraph-codepipelines-ci/pom.xml [hunk 1]
```diff
@@ -65,7 +65,7 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-log4j12</artifactId>
             <version>1.7.12</version>
-            <scope>runtime</scope>
+            <scope>optional</scope>
```
Comment (mbrukman) [inline janusgraph-codepipelines-ci/pom.xml]: Is `<scope>optional</scope>` valid? Was this meant to be `<optional>true</optional>`?

According to [Maven docs](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope), the valid scopes are: `compile`, `provided`, `runtime`, `test`, `system`, and `import`.

Comment (sjudeng) [reply janusgraph-codepipelines-ci/pom.xml]: @mbrukman Yep good catch. I'll push the fix later today.

File: janusgraph-cql/pom.xml [hunk 2]
```diff
@@ -35,13 +35,10 @@
             <groupId>com.datastax.cassandra</groupId>
             <artifactId>cassandra-driver-core</artifactId>
             <version>${cassandra-driver.version}</version>
-            <exclusions>
-                <!-- conflicts with netty-all from org.apache.hbase:hbase-client -->
-                <exclusion>
-                    <groupId>io.netty</groupId>
-                    <artifactId>netty-handler</artifactId>
-                </exclusion>
-            </exclusions>
```
[the remark this answers is not in the document]
Comment (sjudeng) [reply janusgraph-cql/pom.xml]: @pluradj Can you try adding the below to janusgraph-cql POM?

```xml
<dependency>
  <groupId>io.netty</groupId>
  <artifactId>netty-handler</artifactId>
</dependency>
```

[the remark this answers is not in the document]
Comment (sjudeng) [reply janusgraph-cql/pom.xml]: Nevermind, `netty-handler` is already there as an explicit dependency (with version 4.0.50.Final). I'm not able to reproduce the convergence error. I started with pluradj/examples, merged sjudeng/issue468 and then did a clean build (`mvn clean install -DskipTests`) without issue. Can you confirm the issue with a clean build?

[the remark this answers is not in the document]
Comment (pluradj) [reply janusgraph-cql/pom.xml]: You're right, worked fine when I merged both into the same build.

**Comment by pluradj** [comment]
LGTM +1

**Pull request merged**